src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -8,192 +8,156 @@
 
-            /// <summary>
-            /// A task completion source representing the result of this queue item's work.
-            /// This is the task that the client is waiting on.
-            /// </summary>
-            private readonly TaskCompletionSource<TResponseType?> _completionSource = new();
+        /// <summary>
+        /// A task completion source representing the result of this queue item's work.
+        /// This is the task that the client is waiting on.
+        /// </summary>
+        private readonly TaskCompletionSource<TResponseType> _completionSource = new();
 
-            public bool RequiresLSPSolution { get; }
+        public bool RequiresLSPSolution { get; }
 
-            public bool MutatesSolutionState { get; }
+        public bool MutatesSolutionState { get; }
 
-            public string MethodName { get; }
+        public string MethodName { get; }
 
-            public TextDocumentIdentifier? TextDocument { get; }
+        public TextDocumentIdentifier? TextDocument { get; }
 
-            public ClientCapabilities ClientCapabilities { get; }
+        public ClientCapabilities ClientCapabilities { get; }
 
-            public Guid ActivityId { get; }
+        public Guid ActivityId { get; }
 
-            public RequestMetrics Metrics { get; }
+        public IRequestMetrics Metrics { get; }
 
-            public QueueItem(
-                bool mutatesSolutionState,
-                bool requiresLSPSolution,
-                ClientCapabilities clientCapabilities,
-                string methodName,
-                TextDocumentIdentifier? textDocument,
-                TRequestType request,
-                IRequestHandler<TRequestType, TResponseType> handler,
-                Guid activityId,
-                ILspLogger logger,
-                RequestTelemetryLogger telemetryLogger,
-                CancellationToken cancellationToken)
-            {
-                // Set the tcs state to cancelled if the token gets cancelled outside of our callback (for example the server shutting down).
-                cancellationToken.Register(() => _completionSource.TrySetCanceled(cancellationToken));
+        public QueueItem(
+            bool mutatesSolutionState,
+            bool requiresLSPSolution,
+            ClientCapabilities clientCapabilities,
+            string methodName,
+            TextDocumentIdentifier? textDocument,
+            TRequestType request,
+            IRequestHandler<TRequestType, TResponseType, RequestContext> handler,
+            Guid activityId,
+            ILspLogger logger,
+            RequestTelemetryLogger telemetryLogger,
+            CancellationToken cancellationToken)
+        {
+            // Set the tcs state to cancelled if the token gets cancelled outside of our callback (for example the server shutting down).
+            cancellationToken.Register(() => _completionSource.TrySetCanceled(cancellationToken));
```

Comment:
We'll want to capture the token registration and dispose of it later

Reply:
I think this already happens in the caller (queue).  I fixed a bug on that